src/settings_gui.cpp

```diff
@@ -538,7 +540,8 @@ struct GameOptionsWindow : Window {
 
 			case WID_GO_GUI_ZOOM_DROPDOWN:
 				GfxClearSpriteCache();
-				_gui_zoom = (ZoomLevel)(ZOOM_LVL_OUT_4X - index);
+				_gui_zoom_cfg = index > 0 ? ZOOM_LVL_OUT_4X - index + 1 : ZOOM_LVL_GUI_AUTO;
+				UpdateGUIZoom();
```

Comment:
What the old code already didn't, and which is rather annoying from a user experience: when the option doesn't change, it still reloads everything (which hangs the game for a bit). Shouldn't we check here if the value really changed and ignore updates if it does? Possibly a bit outside the current PR, but it is one of those annoying things :P